src/ethereum-json-rpc-client/src/http_outcall.rs

```diff
@@ -2,30 +2,85 @@ use std::future::Future;
 use std::pin::Pin;
 
 use anyhow::Context;
-use ic_exports::ic_cdk::api::management_canister::http_request::{
+use ic_cdk::api::management_canister::http_request::{
     self, CanisterHttpRequestArgument, HttpHeader, HttpMethod, TransformContext,
 };
+#[cfg(feature = "sanitize-http-outcall")]
+use ic_cdk::api::management_canister::http_request::{HttpResponse, TransformArgs};
+use ic_exports::ic_cdk;
 use jsonrpc_core::Request;
 
 use crate::Client;
 
-/// Http outcall client implementation.
+/// EVM client that uses HTTPS Outcalls to communicate with EVM.
+///
+/// This client can be used to connect to external EVM RPC services, but due to IC HTTPS Outcall
+/// protocol interface, there are a few limitations that these services must comply with:
+/// * Identical requests must produce identical response bodies.
+/// * If some of the response headers may vary for different requests, response transformation must
+///   be used (see [HttpOutcallClient::new_with_transform]).
+/// * The service must not use redirects or any other form of indirection. All valid RPC requests
+///   must be answered with `200 OK` status code and a valid RPC response.
+/// * The service must support batched RPC requests.
+///
+/// Note that when a canister is run in replicated mode (e.g. on IC mainneet), every call to
+/// [`HttpClient::send_rpc_request`] will result in multiple concurrent identical HTTP POST requests
+/// to the target EVM.
+///
+/// For more information about how IC HTTPS Outcalls work see [IC documentation](https://internetcomputer.org/docs/current/tutorials/developer-journey/level-3/3.2-https-outcalls/)
 #[derive(Debug, Clone)]
 pub struct HttpOutcallClient {
     url: String,
     max_response_bytes: Option<u64>,
+    transform_context: Option<TransformContext>,
 }
 
 impl HttpOutcallClient {
     /// Creates a new client.
     ///
     /// # Arguments
-    /// * `url` - The url of the canister.
-    ///
+    /// * `url` - the url of the RPC service to connect to.
     pub fn new(url: String) -> Self {
         Self {
             url,
             max_response_bytes: None,
+            transform_context: None,
+        }
+    }
+
+    /// Creates a new client.
+    ///
+    /// You can use [`new_sanitized`] method to use default transform context. (Available with
+    /// Cargo feature `sanitize-http-outcall`)
+    ///
+    /// # Arguments
+    /// * `url` - the url of the RPC service to connect to.
+    /// * `transform_context` - method to use to sanitize HTTP response
+    pub fn new_with_transform(url: String, transform_context: TransformContext) -> Self {
```

Comment:
I would use a constructor method instead `with_transform(mut self, transform_context) -> Self` and the same for `sanitized`